### PR TITLE
sessionctx: add tidb_gogc_tuner_threshold

### LINKED ISSUE #38431: add variable to dynamically change the threshold of GOGC tuner

### PR DESCRIPTION
**Pull request by hawkingrei**
… threshold

Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #38431

Problem Summary:

### What is changed and how it works?

the threshold of GOGC tuner is lower than tidb_server_memory_limit_gc_trigger.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -733,6 +733,36 @@ var defaultSysVars = []*SysVar{
...
+			}
+			return strconv.FormatFloat(floatValue, 'f', -1, 64), nil
+		},
+		SetGlobal: func(s *SessionVars, val string) error {
+			factor, err := strconv.ParseFloat(val, 64)
+			if err != nil {
+				return err
+			}
+			GOGCTunerThreshold.Store(factor)
+			memTotal, err := memory.MemTotal()
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply sessionctx/variable/sysvar.go]: Done

**Comment by hawkingrei** [comment]
@XuHuaiyu @wshwsh12 PTAL

**Review by xhebox — APPROVED** [review]

**Comment by hawkingrei** [comment]
@XuHuaiyu @wshwsh12 PTAL

**Review by wshwsh12 — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**